src/app/core/services/localeService.ts

```diff
@@ -1,24 +1,56 @@
-import { Injectable } from '@angular/core';
+import { Injectable, OnDestroy } from '@angular/core';
+import { BehaviorSubject, Subscription } from 'rxjs';
 import { appConfig } from 'src/app/shared/helpers/config';
+import { UserSessionService } from 'src/app/shared/services/user-session.service';
+
+export interface LanguageMismatch {
+  platformLanguage: string | undefined,
```

Comment:
A platform not having a current language does not really make sense. I realize that comes from currentTag which is optional. We should "fail-fast" and do something with this info directly instead of propagating this "should not happen" everywhere. 
Actually it is also maybe the case of `languages`, I do not remember the point of allowing it not to be defined (including in config).

Reply:
I agree, however the scope is starting to grow somehow, do you mind if I make intermediary PRs ?